client/src/main/java/org/eclipse/hono/client/MessageConsumer.java

```diff
@@ -30,4 +31,13 @@
      * @param closeHandler A handler that is called back with the result of the attempt to close the links.
      */
     void close(Handler<AsyncResult<Void>> closeHandler);
+
+    /**
+     * Grants the given number of message credits to the sender.
+     *
+     * For use when created with 0 prefetch in consumer creation
+     *
+     * @param credits the credits to flow
+     */
+    void flow(int credits);
```

Comment:
IMHO this method should throw a runtime exception if the consumer has been configured for automatic flow control.